<?xml version="1.0"?>

<gazebo:world 
  xmlns:xi="http://www.w3.org/2001/XInclude"
  xmlns:gazebo="http://playerstage.sourceforge.net/gazebo/xmlschema/#gz" 
  xmlns:model="http://playerstage.sourceforge.net/gazebo/xmlschema/#model" 
  xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor" 
  xmlns:window="http://playerstage.sourceforge.net/gazebo/xmlschema/#window" 
  xmlns:param="http://playerstage.sourceforge.net/gazebo/xmlschema/#param" 
  xmlns:body="http://playerstage.sourceforge.net/gazebo/xmlschema/#body" 
  xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom" 
  xmlns:joint="http://playerstage.sourceforge.net/gazebo/xmlschema/#joint" 
  xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" 
  xmlns:ui="http://playerstage.sourceforge.net/gazebo/xmlschema/#ui"
  xmlns:rendering="http://playerstage.sourceforge.net/gazebo/xmlschema/#rendering" 
  xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller"
  xmlns:physics="http://playerstage.sourceforge.net/gazebo/xmlschema/#physics" >

  <verbosity>5</verbosity>

  <physics:ode>
    <stepTime>0.001</stepTime>
    <gravity>0 0 -9.8</gravity>
    <cfm>0.0000000001</cfm>
    <erp>0.2</erp>

    <stepType>quick</stepType>
    <stepIters>100</stepIters>
    <stepW>1.3</stepW>
    <contactMaxCorrectingVel>100.0</contactMaxCorrectingVel>
    <contactSurfaceLayer>0.00</contactSurfaceLayer>

  </physics:ode>

  <rendering:gui>
    <type>fltk</type>
    <size>800 600</size>
    <pos>0 0</pos>
  </rendering:gui>

  <rendering:ogre>
    <shadows>true</shadows>
  </rendering:ogre>

   <!-- Ground Plane -->
  <model:physical name="plane1_model">
    <xyz>0 0 0</xyz>
    <rpy>0 0 0</rpy>
    <static>true</static>

    <body:plane name="plane1_body">
      <geom:plane name="plane1_geom">
        <normal>0 0 1</normal>
        <size>100 100</size>
        <segments>1 1</segments>
        <uvTile>100 100</uvTile>
        <material>Gazebo/GrayGrid</material>
      </geom:plane>
    </body:plane>
  </model:physical>

  <model:physical name='custom_box'>
    <xyz>0.15 0 0.1</xyz>
    <rpy>0 0 0</rpy>
    <body:box name='body'>
      <geom:box name='geom'>
        <size>0.1 0.10 0.2</size>
        <mass>0.01</mass>
        <kp>100000000.0</kp>
        <kd>1.0</kd>
        <bounce>0</bounce>
        <bounceVel>100000</bounceVel>
        <slip1>0.01</slip1>
        <slip2>0.01</slip2>
        <visual>
          <size>0.1 0.1 0.2</size>
          <mesh>unit_box</mesh>
          <material>Gazebo/Rocky</material>
        </visual>
      </geom:box>
    </body:box>
  </model:physical>


  <model:physical name="pioneer_gripper">
    <xyz>0 0 0.0</xyz>
    <rpy>0.0 0.0 0.0</rpy>
  
    <body:box name="gripper_base_body">
      <xyz>0.0 0 0.08</xyz>
      <geom:box name="gripper_base_geom">
        <xyz>0 0 0</xyz>
        <size>0.053 0.123 0.16</size>
        <mass>10.1</mass>
        <visual>
          <size>0.053 0.123 0.16</size>
          <mesh>unit_box</mesh>
          <material>Gazebo/Black</material>
        </visual>
      </geom:box>
    </body:box>
    
    <body:box name="lift_body">
       <xyz>0.061 0 0.0425</xyz>
      <geom:box name="lift_geom">
        <xyz>0 0 0</xyz>
        <size>0.069 0.313 0.063</size>
        <mass>0.1</mass>
        <visual>
          <size>0.069 0.313 0.063</size>
          <mesh>unit_box</mesh>
          <material>Gazebo/Black</material>
        </visual>
      </geom:box>
    </body:box>
    
    <body:box name="left_paddle_body">
      <xyz>0.145 0.135 0.0315</xyz>
      <geom:box name="left_paddle_geom">
        <xyz>0 0 0</xyz>
        <size>0.095 0.025 0.039</size>
        <mass>0.1</mass>
        <visual>
          <size>0.095 0.025 0.039</size>
          <mesh>unit_box</mesh>
          <material>Gazebo/Blue</material>
        </visual>
      </geom:box>
    </body:box>
    
    <body:box name="right_paddle_body">
      <xyz>0.145 -0.135 0.0315</xyz>
      <geom:box name="right_paddle_geom">
        <xyz>0 0 0</xyz>
        <size>0.095 0.025 0.039</size>
        <mass>0.1</mass>
    
        <visual>
          <size>0.095 0.025 0.039</size>
          <mesh>unit_box</mesh>
          <material>Gazebo/Red</material>
        </visual>
      </geom:box>
    </body:box>
    
    <!--<joint:hinge name="fixed">
      <body1>gripper_base_body</body1>
      <body2>world</body2>
      <anchor>gripper_base_body</anchor>
      <anchorOffset>0 0 0.0</anchorOffset>
      <axis>0 0 1</axis>
      <lowStop>0</lowStop>
      <highStop>0</highStop>
      <erp>0.4</erp>
      <cfm>0.01</cfm>
    </joint:hinge>
    -->
    
    <joint:slider name="lift_slider_joint">
      <body1>gripper_base_body</body1>
      <body2>lift_body</body2>
      <anchor>gripper_base_body</anchor>
      <axis>0 0 1</axis>
      <lowStop>-0.05</lowStop>
      <highStop>0.0</highStop>
      <erp>0.3</erp>
      <cfm>10e-5</cfm>
    </joint:slider>
    
    <joint:slider name="left_paddle_joint">
      <body1>lift_body</body1>
      <body2>left_paddle_body</body2>
      <anchor>lift_body</anchor>
      <anchorOffset>0.085 0 0 </anchorOffset>
      <axis>0 1 0</axis>
      <lowStop>0.0</lowStop>
      <highStop>0.12</highStop>
      <erp>0.1</erp>
      <cfm>10e-5</cfm>
    </joint:slider>
    
    <joint:slider name="right_paddle_joint">
      <body1>lift_body</body1>
      <body2>right_paddle_body</body2>
      <anchor>lift_body</anchor>
      <anchorOffset>0.085 0 0 </anchorOffset>
      <axis>0 1 0</axis>
      <lowStop>-0.12</lowStop>
      <highStop>0.0</highStop>
      <erp>0.1</erp>
      <cfm>10e-5</cfm>
    </joint:slider>
  </model:physical>

  <!-- White Directional light -->
  <model:renderable name="directional_white">
    <static>false</static>
    <light>
      <type>directional</type>
      <direction>0 0.05 -0.9</direction>
      <diffuseColor>0.7 0.7 0.7</diffuseColor>
      <specularColor>0.0 0.0 0.0</specularColor>
      <range>100</range>

      <attenuation>1.0 0.01 0.0001</attenuation>
    </light>
  </model:renderable>

</gazebo:world>
